subsys/net/ip/tcp2.c

```diff
@@ -241,6 +252,12 @@ static int tcp_conn_unref(struct tcp *conn)
 
 	tcp_send_queue_flush(conn);
 
+	if (conn->unacked_data) {
+		tcp_pkt_unref(conn->unacked_data);
+		conn->unacked_data = NULL;
+	}
```

Comment:
If we to tcp_pkt_alloc in tcp_conn_alloc(), these lines cound be simplified into just unref().